src/JustEat.StatsD/StatsDMessageFormatter.cs

```diff
@@ -104,34 +104,46 @@ public string Increment(long magnitude, double sampleRate, params string[] statB
         public string Gauge(double magnitude, string statBucket)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g", _prefix, statBucket, magnitude);
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
         public string Gauge(double magnitude, string statBucket, DateTime timestamp)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g|@{3}", _prefix, statBucket, magnitude, timestamp.AsUnixTime());
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Gauge(long magnitude, string statBucket)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g", _prefix, statBucket, magnitude);
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Gauge(long magnitude, string statBucket, DateTime timestamp)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g|@{3}", _prefix, statBucket, magnitude, timestamp.AsUnixTime());
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Event(string name)
         {
             return Increment(name);
         }
 
-        private string Format(String stat, double sampleRate)
+        private string Format(double sampleRate, string stat)
         {
-            return Format(sampleRate, stat);
+            if (sampleRate < DefaultSampleRate)
+            {
+                if (Random.NextDouble() <= sampleRate)
+                {
+                    return string.Format(_cultureInfo, "{0}|@{1:f}", stat, sampleRate);
+                }
+            }
+            else
+            {
+                return stat;
+            }
+
+            return string.Empty;
```

Comment:
At first I thought this `return` was impossible. Maybe put it as an `else` inside the first `if` block instead?